drivers/adc/adc_stm32.c

```diff
@@ -1359,6 +1362,7 @@ static int adc_stm32_channel_setup(const struct device *dev,
 	defined(CONFIG_SOC_SERIES_STM32G0X) ||                                                     \
 	defined(CONFIG_SOC_SERIES_STM32L0X) ||                                                     \
 	defined(CONFIG_SOC_SERIES_STM32U0X) ||                                                     \
+	defined(CONFIG_SOC_SERIES_STM32U3X) ||                                                     \
```

Comment:
To be removed. U3 has no clock configuration inside the driver (it is all in the RCC).